freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/repair/changeprimarygw/action/ChangePrimaryGatewayActions.java

```diff
@@ -150,7 +154,11 @@ protected void doExecute(ChangePrimaryGatewayContext context, StackEvent payload
                 Stack stack = context.getStack();
                 SuccessDetails successDetails = new SuccessDetails(stack.getEnvironmentCrn());
 
-                stackUpdater.updateStackStatus(stack.getId(), getChangePrimaryGatewayCompleteStatus(variables), "Finished changing the primary gateway");
+                if (isFinalChain(variables)) {
+                    stackStatusCheckerJob.syncAStack(stack);
+                } else {
+                    stackUpdater.updateStackStatus(stack.getId(), getChangePrimaryGatewayCompleteStatus(variables), "Finished changing the primary gateway");
```

Comment:
`getChangePrimaryGatewayCompleteStatus` implementation also checks `isFinalChain(variables)` which doesn't make too much sense after this change. most probably you should replace the method call with `DetailedStackStatus.REPAIR_IN_PROGRESS`

Reply:
Thanks! I removed the method getChangePrimaryGatewayCompleteStatus() since it is not necessary and I use DetailedStackStatus.REPAIR_IN_PROGRESS directly. I tested the repair flow again and attached evidence below.